geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/wancommand/ListGatewaysCommandDUnitTest.java

```diff
@@ -369,4 +369,144 @@ public void listGatewaysShouldCorrectlyUpdateSendersConnectedCountWhenReceiverSt
         .hasRowSize(1)
         .hasColumn("Sender Count").containsExactly("0");
   }
+
+  @Test
+  public void testListGatewaySenderOnly() {
+    setupClusters();
+
+    String command = CliStrings.LIST_GATEWAY + " --" + CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY;
+    CommandResultAssert commandAssert = gfsh.executeAndAssertThat(command).statusIsSuccess();
+    commandAssert
+        .hasTableSection("gatewaySenders").hasRowSize(4)
+        .hasColumns().contains("GatewaySender Id", "Member");
+    commandAssert.hasNoSection("gatewayReceivers");
+  }
+
+  @Test
+  public void testListGatewayReceiversOnly() {
+    setupClusters();
+
+    String command = CliStrings.LIST_GATEWAY + " --" + CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY;
+    CommandResultAssert commandAssert = gfsh.executeAndAssertThat(command).statusIsSuccess();
+    commandAssert.hasNoSection("gatewaySenders");
+    commandAssert.hasTableSection("gatewayReceivers")
+        .hasRowSize(1).hasColumns().contains("Port", "Member");
+  }
+
+  @Test
+  public void testListAllGatewaysSendersOnlyFalse() {
+    setupClusters();
+
+    String command =
+        CliStrings.LIST_GATEWAY + " --" + CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY + "=false";
+    CommandResultAssert commandAssert = gfsh.executeAndAssertThat(command).statusIsSuccess();
+    commandAssert
+        .hasTableSection("gatewaySenders").hasRowSize(4)
+        .hasColumns().contains("GatewaySender Id", "Member");
+    commandAssert.hasTableSection("gatewayReceivers")
+        .hasRowSize(1).hasColumns().contains("Port", "Member");
+  }
+
+  @Test
+  public void testListAllGatewaysReceiversOnlyFalse() {
+    setupClusters();
+
+    String command =
+        CliStrings.LIST_GATEWAY + " --" + CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY + "=false";
+    CommandResultAssert commandAssert = gfsh.executeAndAssertThat(command).statusIsSuccess();
+    commandAssert
+        .hasTableSection("gatewaySenders").hasRowSize(4)
+        .hasColumns().contains("GatewaySender Id", "Member");
+    commandAssert.hasTableSection("gatewayReceivers")
+        .hasRowSize(1).hasColumns().contains("Port", "Member");
+  }
+
+  @Test
+  public void testListAllGatewaysBothFalse() {
```

Comment:
3 of these 4 test cases are redundant and can be safely removed without losing coverage, since the behaviour being tested is just whether the options `--senders-only` and `--receivers-only` are specified at all, not what their values are. If one of these tests were to fail but not the other 3, that would indicate a deeper problem with the gfsh parser that is outside the scope of what this class is intended to be testing and should be already covered in an existing test class.